style: general styling improvements

nicolellou opened the pull request.

nicolellou committed:
style: general styling improvements

ksamson2 commented on src/components/Landing/Landing.tsx:
```diff
@@ -21,20 +21,37 @@ const Landing: React.FC<Props> = ({ title, subtitle }) => {
+        <div className="is-flex is-flex-direction-column is-justify-content-center is-align-items-center">
+          <div className="media">
+            <div className="block">
+              <i>Take or upload photo</i>
```
hmm I don't think we should use the `<i>` tag. Can we access this in a css style instead?

ksamson2 commented on src/components/PhotoAnalysis/PhotoAnalysis.tsx:
```diff
@@ -80,12 +83,15 @@ const Analysis: React.FC = () => {
       <br />
-      <img src={imageSrc} alt="user's uploaded file" />
+
+      <img src={imageSrc} />
```
If I merge my changes in before this CR is complete, it would be worth it to rebase first and then update this PR

ksamson2 reviewed (dismissed):
I left a small comment to update. This is non-blocking though, so you can merge this in and address the comment in a later PR

nicolellou committed:
style: image style improvements

nicolellou committed:
Merge branch 'develop' into photoanalysis-ui

ksamson2 commented on src/components/PhotoAnalysis/PhotoAnalysis.tsx:
```diff
@@ -50,6 +53,31 @@ const Analysis: React.FC = () => {
       .catch(err => setRekognitionResponse(JSON.stringify(err, null, 2)))
   }
 
+  const displayUploadButton = () => {
```
@christoff-linde @brian-david @Jaceyjay 
I helped Nicole with this part, so I just wanted another opinion here. Any strong objections to hiding/showing the upload button in this way? I'm sure there are multiple ways to achieve this functionality; this just seemed to be the simplest

christoff-linde replied:
IMHO it is a bit clunky to read, but that might just be me?

ksamson2 replied:
That's fine... how do you suggest handling this case?

christoff-linde replied:
Eh, we can just merge it 😃 too much brain power required to rethink

ksamson2 reviewed (dismissed):
Looks good to me (LGTM)

ksamson2 commented on src/components/PhotoAnalysis/PhotoAnalysis.tsx:
```diff
@@ -50,6 +53,31 @@ const Analysis: React.FC = () => {
+                  className="file-input"
+                  type="file"
+                  onChange={identifyImageLabels}
+                ></input>
```
Small comment here (non-blocking)... this doesn't need a closing tag 🤔 

We could just say: 
```
 <input
      className="file-input"
      type="file"
      onChange={identifyImageLabels}
   />
```

christoff-linde committed:
refactor: remove extra input closing tag

christoff-linde committed:
refactor: remove eslint disable rule & swap image to StaticImage

christoff-linde committed:
refactor: remove StaticImage & change back to html img

With Gatsby, images are usually stored either in local storage (for e.g., the static folder) or on a hosted service/CMS. This requires images to be either 'hard coded' using StaticImage or for images with 'dynamic' sources, to be rendered using GatsbyImage. This requires the image url to be known at build time.

At this stage, it is unclear exactly what workaround would be needed for images that are added/created at run time in the browser. Thus, it would be easier to just switch back to using standard HTML img

christoff-linde committed:
refactor: hide TextToSpeech controls if no labels have been generated

christoff-linde commented on src/components/PhotoAnalysis/PhotoAnalysis.tsx:
```diff
@@ -50,10 +53,36 @@ const Analysis: React.FC = () => {
   }
 
+  const displayUploadButton = () => {
+    if (!isLoading && rekognitionResponse === "")
```
We can't check for `rekognitionResponse` here, because it causes the upload button to hide after an image has been uploaded. Which is probably intended, but it has the side effect of requiring a user to refresh the page if they want to upload another image

ksamson2 replied:
This was intentional. There will be another button at the top of the screen that allows the user to upload a new image and it will trigger the same functionality. We can talk about placement/design of that button but I don't think it affects the current task or this function

christoff-linde commented on src/components/PhotoAnalysis/PhotoAnalysis.tsx:
```diff
@@ -64,32 +93,39 @@ const Analysis: React.FC = () => {
-      <img src={imageSrc} alt="user's uploaded file" />
-      {processRekognitionLabels(rekognitionResponse as IdentifyLabelsOutput)}
-      <p>{labels.join(", ")}</p>
+      {imageSrc !== undefined ? (
+        <div className="card-image">
+          <img src={imageSrc} alt="Uploaded preview" />
+        </div>
+      ) : null}
```
The image needs to be displayed inside the `card-content` (or between the _take or upload photo_ text and the actual upload button)

christoff-linde commented on src/components/PhotoAnalysis/PhotoAnalysis.scss:
```diff
@@ -0,0 +1,27 @@
+
+.cameraButton{
+  position: relative;
+  top: 425px;
```
This will need to be changed. We need to find a way to position the `cameraButton` at the bottom regardless of the size of the content in the card (i.e. not with a fixed `px` value from the top). Maybe actually try `bottom`?

ksamson2 replied:
Wouldn't a media query help this when we get to the responsive design section? (as in, can we keep the somewhat fixed positioning and handle responsiveness in a grid?)

nicolellou committed:
style: fixed button and card

christoff-linde reviewed (approved):
Noice
